contrib/endpoints/src/api_manager/context/service_context.cc

```diff
@@ -76,6 +76,18 @@ ServiceContext::ServiceContext(std::unique_ptr<ApiManagerEnvInterface> env,
 
   service_account_token_.SetAudience(
       auth::ServiceAccountToken::JWT_TOKEN_FOR_FIREBASE, kFirebaseAudience);
+
+  if (config_->server_config() &&
+      !config_->server_config()
+           ->api_check_security_rules_config()
+           .authorization_service_audience()
+           .empty()) {
+    service_account_token_.SetAudience(
+        auth::ServiceAccountToken::JWT_TOKEN_FOR_AUTHORIZATION_SERVICE,
```

Comment:
"aud" is a standard JWT claim. It's not ESP specific. Any service that accepts JWT defines what audience it accepts. Google services accept JWT audience that matches the service name. Not all services do the same, and the service name can have different formats. That's why we need to ask API producers to specify the accepted audience in the function specification when they write rules. 

I am ok with having this change in a separate PR.